src/infer_request.h

```diff
@@ -69,8 +73,14 @@ class InferRequest {
   const std::set<std::string>& RequestedOutputNames();
   bi::managed_external_buffer::handle_t ShmHandle();
 
+  void SetPrevPromise(std::promise<std::unique_ptr<InferResponse>>** promise);
+  void SetValueForPrevPromise(std::unique_ptr<InferResponse> infer_response);
```

Comment:
use const& for `infe_response`

Reply:
Here we are calling `set_value` to set the promise with the `infer_response` in this function. The ownership of `infer_response` is transferring, I might be wrong here but I think due to that we cannot use `const&` for `infer_response`.